git/client.go

```diff
@@ -581,11 +586,25 @@ func (c *Client) HaveObject(id Sha1) (found bool, packedfile File, err error) {
 			if v2PackIndexHasSha1(c, pfile, buf, id) {
 				// We want to return the pack file, not the index.
 				f.Close()
+				log.Printf("Found object %s in pack file %s\n", id, fi.Name())
 				return true, pfile, nil
 			}
 			f.Close()
 		}
 	}
+
+	gitdirsnapshot := ""
+	filepath.Walk(string(c.GitDir), func(path string, info os.FileInfo, err error) error {
+		gitdirsnapshot = gitdirsnapshot + fmt.Sprintf(" %s:%d ", path, info.Size())
+		return nil
+	})
+
+	log.Printf("None of the pack files has object %s\nSnapshot: %s\n", id, gitdirsnapshot)
+	branches, _ := c.GetBranches()
+	for _, branch := range branches {
+		cmt1, _ := branch.CommitID(c)
+		log.Printf("Branch: %s -> %s\n", branch.String(), cmt1)
+	}
```

Comment:
Do we need all these log statements (and the walk above)? This seems like a lot of overhead to add for something that's only there in case DGIT_TRACE is set.

Reply:
I'm removing these shortly. Sorry, I put in a whole bunch of them just to narrow down the intermittent problem I was seeing.

Reply:
That's what I figured.. let me know when it's ready, because overall I think this is ready to merge.